From: "Matthew Wilcox (Oracle)" <willy@infradead.org>
Subject: mm: remove try_to_free_swap()
Date: Fri, 2 Sep 2022 20:46:44 +0100

All callers have now been converted to folio_free_swap() and we can remove
this wrapper.

Link: https://lkml.kernel.org/r/20220902194653.1739778-49-willy@infradead.org
Signed-off-by: Matthew Wilcox (Oracle) <willy@infradead.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/swap.h |    6 ------
 mm/folio-compat.c    |    7 -------
 mm/memory.c          |    2 +-
 3 files changed, 1 insertion(+), 14 deletions(-)

--- a/include/linux/swap.h~mm-remove-try_to_free_swap
+++ a/include/linux/swap.h
@@ -510,7 +510,6 @@ extern int __swp_swapcount(swp_entry_t e
 extern int swp_swapcount(swp_entry_t entry);
 extern struct swap_info_struct *page_swap_info(struct page *);
 extern struct swap_info_struct *swp_swap_info(swp_entry_t entry);
-extern int try_to_free_swap(struct page *);
 struct backing_dev_info;
 extern int init_swap_address_space(unsigned int type, unsigned long nr_pages);
 extern void exit_swap_address_space(unsigned int type);
@@ -594,11 +593,6 @@ static inline int swp_swapcount(swp_entr
 {
 	return 0;
 }
-
-static inline int try_to_free_swap(struct page *page)
-{
-	return 0;
-}
 
 static inline swp_entry_t folio_alloc_swap(struct folio *folio)
 {
--- a/mm/folio-compat.c~mm-remove-try_to_free_swap
+++ a/mm/folio-compat.c
@@ -146,10 +146,3 @@ void putback_lru_page(struct page *page)
 {
 	folio_putback_lru(page_folio(page));
 }
-
-#ifdef CONFIG_SWAP
-int try_to_free_swap(struct page *page)
-{
-	return folio_free_swap(page_folio(page));
-}
-#endif
--- a/mm/memory.c~mm-remove-try_to_free_swap
+++ a/mm/memory.c
@@ -3844,7 +3844,7 @@ vm_fault_t do_swap_page(struct vm_fault
 
 	if (swapcache) {
 		/*
-		 * Make sure try_to_free_swap or swapoff did not release the
+		 * Make sure folio_free_swap() or swapoff did not release the
 		 * swapcache from under us.  The page pin, and pte_same test
 		 * below, are not enough to exclude that.  Even if it is still
 		 * swapcache, we need to check that the page's swap has not
_
